aqueduct/management/admin.py

```diff
@@ -52,11 +62,45 @@ def clean_excluded_models(self):
         return self.cleaned_data["excluded_models"]
 
 
-class UserProfileAdminForm(ExcludedModelsAdminForm):
+class ExcludedMCPServersAdminForm(forms.ModelForm):
+    class Meta:
+        model = None  # To be specified by subclass
+        fields = "__all__"
+
+    excluded_mcp_servers = forms.MultipleChoiceField(
+        choices=[],
+        required=False,
+        widget=forms.CheckboxSelectMultiple(),
+        label="Excluded MCP Servers",
+        help_text="Choose MCP servers to be excluded.",
+    )
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.fields["excluded_mcp_servers"].choices = [(s, s) for s in get_mcp_server_choices()]
+        if self.instance and getattr(self.instance, "excluded_mcp_servers", None):
+            self.initial["excluded_mcp_servers"] = self.instance.excluded_mcp_servers
+
+    def clean_excluded_mcp_servers(self):
+        # Store as list/JSON, not string
+        return self.cleaned_data["excluded_mcp_servers"]
+
+
+class UserProfileAdminForm(ExcludedModelsAdminForm, ExcludedMCPServersAdminForm):
     class Meta(ExcludedModelsAdminForm.Meta):
         model = UserProfile
         fields = "__all__"
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.fields["excluded_models"].choices = [(m, m) for m in get_model_choices()]
+        if self.instance and getattr(self.instance, "excluded_models", None):
+            self.initial["excluded_models"] = self.instance.excluded_models
```

Comment:
Actually, isn't this piece of code also duplicated - in `ExcludedModelsAdminForm.__init__()`?